From 2c86c77dc5795a7f1e8f915c138c27893d8141d4 Mon Sep 17 00:00:00 2001
Message-ID: <2c86c77dc5795a7f1e8f915c138c27893d8141d4.1736748624.git.sam@gentoo.org>
In-Reply-To: <9f57274faacf1dbb6c2b56d29de150b5e79ecb3d.1736748624.git.sam@gentoo.org>
References: <9f57274faacf1dbb6c2b56d29de150b5e79ecb3d.1736748624.git.sam@gentoo.org>
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Sat, 11 Jan 2025 07:23:27 +0800
Subject: [PATCH 7/8] ld: Document mixing LTO and non-LTO objects for -r

	* ld.texi: Document mixing LTO and non-LTO relocatable files for
	"ld -r".

Signed-off-by: H.J. Lu <hjl.tools@gmail.com>
---
 ld/ld.texi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/ld/ld.texi b/ld/ld.texi
index da714a20855..d1787453f57 100644
--- a/ld/ld.texi
+++ b/ld/ld.texi
@@ -1126,6 +1126,18 @@ relocations.  Different output formats can have further restrictions; for
 example some @code{a.out}-based formats do not support partial linking
 with input files in other formats at all.
 
+On ELF platforms, when the relocatable output contains both contents
+which require link-time optimization (LTO) and contents which don't
+require LTO, a .gnu_object_only section will be created to contain a
+relocatable object file, as if @samp{-r} is applied to all relocatable
+inputs which don't require LTO.  When processing a relocatable input
+with a .gnu_object_only section, the linker will extract the
+.gnu_object_only section as a separate input.
+
+Note that since @samp{-r} groups some sections from different input files
+together, there may be negative impacts on code size and locality in
+final executable or shared library.
+
 This option does the same thing as @samp{-i}.
 
 @kindex -R @var{file}
-- 
2.48.0

